core/utils.py

```diff
@@ -63,13 +63,37 @@ def filter_validity(arg='validity', **kwargs):
     )
 
 
-def filter_validity_business_model(arg='validity', **kwargs):
-    validity = kwargs.get(arg)
-    if validity is None:
-        validity = core.datetime.datetime.now()
+def filter_validity_business_model(arg='dateValidFrom__Gte', arg2='dateValidTo__Lte', **kwargs):
+    date_valid_from = kwargs.get(arg)
+    date_valid_to = kwargs.get(arg2)
+    #default scenario
+    if not date_valid_from and not date_valid_to:
+        today = core.datetime.datetime.now()
+        return __place_the_filters(date_start=today, date_end=today)
+
+    # scenario - only date valid to set
+    if not date_valid_from and date_valid_to:
+        today = core.datetime.datetime.now()
+        oldest = min([today, date_valid_to])
+        return __place_the_filters(date_start=oldest, date_end=oldest)
+
+    # scenario - only date valid from
+    if date_valid_from and not date_valid_to:
+        return __place_the_filters(date_start=date_valid_from, date_end=date_valid_from)
```

Comment:
return __place_the_filters(date_start=date_valid_from, date_end=None)

Reply:
@delcroip I think that it doesn't work if the functions accept such params value, example below:
```
return (
        Q(date_valid_from=None) | Q(date_valid_from__lte=None[date_end]),
        Q(date_valid_to=None) | Q(date_valid_to__gte=date_start)
    )
```
Why?
Expression with lte/gte in django doesn't accept None/Null values.
But it will work if I add special condition in function place the filter:
```
def __place_the_filters(date_start, date_end):
    """funtion related to 'filter_validity_business_model'
    function so as to set up the chosen filters
    to filter the validity of the entity
    """
    if not date_end:
        return (
            Q(date_valid_from=None) | Q(date_valid_from__lte=date_start),
            Q(date_valid_to=None) | Q(date_valid_to__gte=date_start)
        )
    return (
        Q(date_valid_from=None) | Q(date_valid_from__lte=date_end),
        Q(date_valid_to=None) | Q(date_valid_to__gte=date_start)
    )
```